src/bin/pg_autoctl/service_keeper.c

```diff
@@ -372,13 +372,21 @@ keeper_node_active_loop(Keeper *keeper, pid_t start_pid)
 
 		couldContactMonitor = couldContactMonitorThisRound;
 
-		if (couldContactMonitor &&
-			keeperState->assigned_role != keeperState->current_role)
+		if (keeperState->assigned_role != keeperState->current_role)
 		{
 			needStateChange = true;
 
-			log_info("Monitor assigned new state \"%s\"",
-					 NodeStateToString(keeperState->assigned_role));
+			if (couldContactMonitor)
+			{
+				log_info("Monitor assigned new state \"%s\"",
+						 NodeStateToString(keeperState->assigned_role));
+			}
+			else
+			{
+				/* if network is not healthy we might self-assign a state */
```

Comment:
This comment is not strictly correct, since `couldContactMonitor` can be false because either `keeper_refresh_other_nodes`, `keeper_config_set_groupId_and_slot_name` or `keeper_ensure_configuration` failed.

Reply:
To be clear, in that case the new value would come from the monitor. Just some work after it would have failed.

Reply:
That's right. That said the comment is still valid, it's just incomplete. The other cases you mention are that we could contact the monitor, but then failed to sync the new values received with our local state/config. In my notes I have also seen that we should remove the replication slot name from the config, as we now automate it from the node id. We rely in more and more places on the name being `pgautofailover_standby_%d`. So I will open a PR to clean that up and include some more comments on why we don't want to apply the monitor state change when we failed to sync local state or local config from the given values.